server/api.go

```diff
@@ -931,3 +942,124 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request) {
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getRepositories(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	ctx := context.Background()
+	githubClient := p.githubConnect(*info.Token)
+
+	repositories, _, apiErr := githubClient.Repositories.List(ctx, "", &github.RepositoryListOptions{})
+	if apiErr != nil {
+		mlog.Error(apiErr.Error())
+		return
+	}
+
+	resp, _ := json.Marshal(repositories)
```

Comment:
How big does the response to the client get if a user has access to a lot (>400) repos?

Reply:
@jfrerich Do you have any comments on this?

Reply:
I'm not sure how to measure this or what the commonly used solution is to this situation. Is it typical to limit the number of items in a request and if so, how do we determine the limit?

Reply:
It's probably fine since we are doing a one-time fetch. We wouldn't want to skip out on results that are supposed to be there, so I think we shouldn't limit.

What do the results look like the plugin is not set up to be scoped to an org? Do I get only repos that I interact with on GitHub?